rhino/src/main/java/org/mozilla/javascript/ScriptRuntimeES6.java

```diff
@@ -46,4 +46,15 @@ public static void addSymbolUnscopables(
         ScriptableObject.putProperty(unScopablesDescriptor, "writable", false);
         constructor.defineOwnProperty(cx, SymbolKey.UNSCOPABLES, unScopablesDescriptor, false);
     }
+
+    /** Registers the symbol <code>[Symbol.hasInstance]</code> on the given constructor function. */
+    public static void addSymbolHasInstance(
```

Comment:
I get what you're trying to do here, but I don't think that we should add a new function to a "ScriptRuntime" class that we're only going to use in one place. I'd suggest that instead of all this, you just call "defineProperty" on the "constructor" object back in BaseFunction and pass it the bitset (I'll put a comment there too).